test/jdk/java/awt/Window/WindowIconUpdateOnDPIChanging/WindowIconUpdateOnDPIChangingTest.java

```diff
@@ -73,10 +74,11 @@
             + "</html>\n";
 
     public static void main(String[] args) throws Exception {
-        countDownLatch = new CountDownLatch(1);
         SwingUtilities.invokeLater(WindowIconUpdateOnDPIChangingTest::createUI);
-        countDownLatch.await(15, TimeUnit.MINUTES);
-        if (!testResult) {
+        if (!countDownLatch.await(15, TimeUnit.MINUTES)) {
+            frame.dispose();
```

Comment:
It must be called on EDT.
```suggestion
            SwingUtilities.invokeAndWait(() -> frame.dispose());
```

Reply:
That's just a test, do we really need this level of refinement?